### PR TITLE
2243 clarify transpile.md

### PR DESCRIPTION
**Pull request by michael-lloyd-morris**
### 🤔 What's changed?

Clarified the differences between transpiling Cucumber in CommonJS and ESM

### ⚡️ What's your motivation? 

Spent 4 hours slogging through this.

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)

### ♻️ Anything particular you want feedback on?

Is this rewrite clear to others?

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by Jeff-Tian** [comment]
Thanks, @michael-lloyd-morris! Great work, it'll benefit many others unfamiliar with nodejs and esm etc.

**Review by Jeff-Tian — APPROVED** [review]
Looks good to me! Very helpful messages indeeded.

File: docs/transpiling.md [hunk 1]
```diff
@@ -31,10 +33,27 @@ If you are using [ts-node](https://github.com/TypeStrong/ts-node):
...
+
+To use ESM modules set the `"type": "module"` in the package.json file.
 
-```shell
-$ NODE_OPTIONS="--loader ts-node/esm" cucumber-js --import 'step-definitions/**/*.ts'
+Your cucumber.js file will need to be an ESM module and contain the following at a minimum:
+
+```javascript
+export default {
+  requireModule: ['ts-node/register'],
```
Comment (davidjgoss) [inline docs/transpiling.md:54]: This doesn’t seem right? require-module doesn’t affect import

File: docs/transpiling.md [hunk 2]
```diff
@@ -31,10 +33,27 @@ If you are using [ts-node](https://github.com/TypeStrong/ts-node):
...
+    "test": "cross-env NODE_OPTIONS=\"--loader ts-node/esm\" cucumber-js"
+  }
+}
+```
+
+To use ESM modules set the `"type": "module"` in the package.json file.
 
-```shell
-$ NODE_OPTIONS="--loader ts-node/esm" cucumber-js --import 'step-definitions/**/*.ts'
+Your cucumber.js file will need to be an ESM module and contain the following at a minimum:
```
Comment (davidjgoss) [inline docs/transpiling.md:50]: The config file can still be JSON or YAML - I guess we want to say that if JavaScript then it should be the appropriate format.

File: docs/transpiling.md (hunk 1 above)
Comment (davidjgoss) [reply docs/transpiling.md:54]: I think this can just be removed.

File: docs/transpiling.md [hunk 3]
```diff
@@ -31,10 +33,27 @@ If you are using [ts-node](https://github.com/TypeStrong/ts-node):
...
 #### ESM
 
-For ESM projects, you can use `ts-node`'s ESM loader and then `import` your TypeScript files:
+For ESM projects, you can use `ts-node`'s ESM loader and then import your TypeScript files. Doing this will require setting an environment variable. the cleanest way to do this is to include the [cross-env](https://www.npmjs.com/package/cross-env) package with `npm i -D cross-env`
+
+Then change your npm test script invocation in the package.json file to:
+
+```json
+{
+  "scripts: {
```
Comment (davidjgoss) [inline docs/transpiling.md:42]: Missing a double quote on the key here

Comment (michael-lloyd-morris) [reply docs/transpiling.md]: Ah.  Sorry.  Also spotted an uncapitalized T at the start of a sentence so I sent the whole thing through a spell/grammar checker to be sure. Should be good now.

File: docs/transpiling.md (hunk 2 above)
Comment (michael-lloyd-morris) [reply docs/transpiling.md]: Lines removed.

File: docs/transpiling.md (hunk 1 above)
Comment (michael-lloyd-morris) [reply docs/transpiling.md]: It needs to be reworded heavily and I've done so. If you read that header as applying to all ESM projects others will too. That sub header is level 4 and it was meant to cover "with TS-Node to load ES modules"

**Comment by michael-lloyd-morris** [comment]
These examples are fairly complex and they also raise a lot of Google and Bing search results. I have a solution, but it will affect more than this - setup a group of Cucumber-JS scenario repositories that address the most common applications of the system one at a time.

**Comment by michael-lloyd-morris** [comment]
@davidjgoss Please let me know what else do I need to do for this.

File: docs/transpiling.md [hunk 4]
```diff
@@ -1,52 +1,53 @@
...
 "allowSyntheticDefaultImports": true,
 "resolveJsonModule": true,
+
 ```
 
-Other than that, a pretty standard TypeScript setup should work as expected.
+Typescript's output must match Cucumber's expected import, and this is controlled by the "module" option in the `tsconfig.json`.  The default is "CommonJS", but if your project is setup to be an ESM project then Typescript will need to output some generation of ES code to Cucumber. If you aren't sure use "ESNext".
 
-You'll also need to specify where your support code is, since `.ts` files won't be picked up by default.
+Cucumber doesn't load `*.ts` files by default. The glob pattern `step-definitions/**/*.ts` will load them from the default step definition location.
```
Comment (davidjgoss) [inline docs/transpiling.md:24]: I don't think this line is an improvement. `step-definitions` isn't any of our default paths.

File: docs/transpiling.md [hunk 5]
```diff
@@ -1,52 +1,53 @@
...
 ### With ts-node
 
-If you are using [ts-node](https://github.com/TypeStrong/ts-node):
+[TS-Node](https://github.com/TypeStrong/ts-node): is the one of the most popular ways to load TypeScript files. 
 
 - In a configuration file `{ requireModule: ['ts-node/register'], require: ['step-definitions/**/*.ts'] }`
 - On the CLI `$ cucumber-js --require-module ts-node/register --require 'step-definitions/**/*.ts'`
 
-#### ESM
+If you are using ts-node in a CommonJS project then this configuration will work, but if you have an ESM project there are additional steps.
```
Comment (davidjgoss) [inline docs/transpiling.md:35]: We should make clear that both `requireModule` and `require` are not suitable for ESM and that the `--loader` flag and `import` option are their respective equivalents.

File: docs/transpiling.md [hunk 6]
```diff
@@ -1,52 +1,53 @@
...
-For ESM projects, you can use `ts-node`'s ESM loader and then `import` your TypeScript files:
+* Set TypeScript to export to an ES format such as "ESNext" using the `ts-config.json` file. 
+* Then use `ts-node`'s ESM loader to import your TypeScript. 
 
-```shell
-$ NODE_OPTIONS="--loader ts-node/esm" cucumber-js --import 'step-definitions/**/*.ts'
-```
-
-Don't forget to set your `tsconfig.json` to emit JavaScript with `import` and `export` statements:
+That last step requires setting an environment variable. The cleanest way to do this is to include the [cross-env](https://www.npmjs.com/package/cross-env) package with `npm i -D cross-env`. With that package installed make the following change to your npm test script invocation in the package.json file:
```
Comment (davidjgoss) [inline docs/transpiling.md:40]: I don't think we should be prescriptive of using this particular library/approach. By all means mention it as an option but examples should be as vanilla as possible really.

Comment (michael-lloyd-morris) [reply docs/transpiling.md]: Done.

File: docs/transpiling.md (hunk 5 above)
Comment (michael-lloyd-morris) [reply docs/transpiling.md]: Done.

**Review by davidjgoss — APPROVED** [review]

**Pull request merged**